src/compiler/utilities.ts

```diff
@@ -1022,6 +1022,88 @@ namespace ts {
         return node.kind === SyntaxKind.ImportEqualsDeclaration && (<ImportEqualsDeclaration>node).moduleReference.kind !== SyntaxKind.ExternalModuleReference;
     }
 
+    export function isSourceFileJavaScript(file: SourceFile): boolean {
+        return isInJavaScriptFile(file);
+    }
+
+    function isInJavaScriptFile(node: Node): boolean {
+        return !!(node.parserContextFlags & ParserContextFlags.JavaScriptFile);
+    }
+
+    function isCalledToNamedFunction(expression: Node, name: string): boolean;
+    function isCalledToNamedFunction(expression: CallExpression, name: string) {

```

Comment:
I'd prefer not to have an overload but instead write the type assertions explicitly. That's what we do everywhere else.